docs/tasks/administer-cluster/kubeadm-upgrade-ha.md

```diff
@@ -0,0 +1,179 @@
+
+At this point all the static pod manifests in your cluster, i.e. API Server, Controller Manager, Scheduler, Kube Proxy have been upgraded, however the base software, i.e. `kubelet`, `kubectl`, `kubeadm` installed on your nodes’ OS are still of the old version. For upgrading the base software packages on all nodes they will be taken out (`cordon` if master, `drain` if worker node) upgraded and then taken in (`uncordon`) again one by one. 
+
+First, the master nodes the master nodes will be upgraded (run `kubectl cordon` / `kubectl uncordon` from whatever machine you have set up to issue `kubectl` commands):
+
+```shell
+$ kubectl cordon <MASTER-NODE-FQDN>
+$ systemctl stop keepalived kubelet docker
+# use your distro's package manager, e.g. 'yum' on RH-based systems
+# for the versions stick to kubeadm's output (see above)
+$ yum install -y kubelet-<NEW-K8S-VERSION> kubectl-<NEW-K8S-VERSION> kubeadm-<NEW-K8S-VERSION> kubernetes-cni-<NEW-CNI-VERSION>
+$ systemctl start `echo keepalived kubelet docker | tac`
+$ kubectl uncordon <MASTER-NODE-FQDN>
+```
+
+In this example an _rpm_-based system is assumed and `yum` is used for installing the upgraded software. On _deb_-based systems it will be `apt-get update` and then `apt-get install <PACKAGE>=<NEW-K8S-VERSION>` for all packages.
+
+Because on the worker nodes there is no keepalived to stop and start and all processes need to be moved to other nodes cleanly before shutting anything down, `drain` is used instead of `cordon`. In this process all containers will get moved from one node to another at least once. If some of them keep local data you will need some strategy for this scenario. This really depends on the application(s) running in the cluster, and there is no simple, generic solution to this.
+
+On the master (or wherever you run `kubectl` commands):
+
+```shell
+$ kubectl drain --delete-local-data --ignore-daemonsets <WORKER-NODE-FQDN>
+```
+
+On the respective worker node:
+
+```shell
+$ systemctl stop keepalived kubelet docker
+# use your distro's package manager, e.g. 'yum' on RH-based systems
+# for the versions stick to kubeadm's output (see above)
+$ yum install -y kubelet-<NEW-K8S-VERSION> kubectl-<NEW-K8S-VERSION> kubeadm-<NEW-K8S-VERSION> kubernetes-cni-<NEW-CNI-VERSION>
+$ systemctl start `echo keepalived kubelet docker | tac`
+```
+
+Like above, for _deb_-based systems use `apt-get install <PACKAGE>=version` to upgrade the packages to the respective desired versions.
+
+Now the new version of the `kubelet` should be running on the host. Verify this using the following command on the respective host:
+
+```shell
+systemctl status kubelet
+```
+
+If everything was successful, again on the master (or wherever you run `kubectl` commands):
+
+```shell
+$ kubectl uncordon <WORKER-NODE-FQDN>
+```
+
+After all hosts in your cluster have been upgraded, verify that all nodes are available again by executing the following from wherever you run `kubectl` commands:
```

Comment:
Should we instruct the user to check the node status after each node rather than waiting until all nodes have been completed? That would give the opportunity to try and rectify issues with less service disruption.

Reply:
Agreed. Changed the wording.